Character.cpp

```diff
@@ -98,20 +97,38 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        levelUpBonus( hitPoints, initialHitPoints );
+        levelUpBonus( armor, initialArmorLevel );
+        levelUpBonus( attackDamage, initialAttackDamage );
+
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;
     }
 }
 
 void Character::printStats()
 {
     std::cout << getName() << "'s stats: " << std::endl;
-    assert(false);
     /*
     make your getStats() use a function from the Utility.h
     */
     std::cout << getStats(); 
     
     std::cout << std::endl;
     std::cout << std::endl;
-}
\ No newline at end of file
+}
+
+void Character::levelUpBonus( int& statToIncrement, std::unique_ptr<int>& initialValueOfStatToIncrement )
+{
+    // a
+    if( statToIncrement < *initialValueOfStatToIncrement ) 
+    {
+        statToIncrement = *initialValueOfStatToIncrement;
+    } 
+
+    // b
+    statToIncrement += statToIncrement * 0.1;
+    
+    // c
+    *initialValueOfStatToIncrement = statToIncrement;
+    
+}
```

Comment:
This doesn't need to be a member function.  
It can be a free function.  Declare/define it above where it gets used. 
Also, eliminate the `std::unique_ptr<int>&` parameter by making the 2nd argument an `int&`
- [ ] revise